gc/gc-iceberg-files/build.gradle.kts

```diff
@@ -29,6 +29,9 @@ description =
     "due to the hugely different set of dependencies"
 
 dependencies {
+
+  // hadoop-common brings Jackson in ancient versions, pulling in the Jackson BOM to avoid that
+  implementation(platform(libs.jackson.bom))
```

Comment:
wouldnt there be a more central way of enforcing this instead of doing it everywhere where `libs.hadoop.common` gets depended on ?
i.e. https://docs.gradle.org/current/userguide/dependency_version_alignment.html#sec:align_bom_platform

this might also have the benefit of upgrading jackson to our exact version in case it comes in through other transitive dependencies as well (not just hadoop-common)